### PR TITLE
refresh SearcherManagers after partial reindex

### LINKED ISSUE #1187: signal to the webapp once partial reindex is done
Once the configuration listener protocol is extended so that allows more than just Configuration object to be passed, the protocol should be further extended so that partial reindex notifies the webapp to call maybeRefresh() on **specific** SearcherManager objects. This is a follow-up to the changes for #1132.

### PR DESCRIPTION
**Pull request by vladak**
fixes #1187

### CODE REVIEW

**Comment by vladak** [comment]
Not entirely sure how to test this.

This would probably work however is a bit heavyweight:
  - disable periodic refreshing of searcherManagers
  - full index of a repo
  - start webapp listener
  - perform search (so that searcherManager is created)
    - either the whole webapp needs to be running or use the index files directly
  - modify the repo
  - partial index
  - perform search for the changes done with the most recent modification

File: src/org/opensolaris/opengrok/configuration/messages/RefreshMessage.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * Message for refreshing SearcherManagers in the webapp after partial reindex.
+ * The list of projects is carried in tags.
+ *
+ * @author Vladimir Kotal
+ */
+public class RefreshMessage extends Message {
+    @Override
+    public byte[] apply(RuntimeEnvironment env) throws IOException {
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/configuration/messages/RefreshMessage.java]: Did you get this to compile with merged master? `apply` now is marked as `final`. The method you should override is `applyMessage`

File: src/org/opensolaris/opengrok/configuration/RuntimeEnvironment.java [hunk 2]
```diff
@@ -1172,6 +1172,26 @@ protected void writeConfiguration() throws IOException {
...
+     * This is used for partial reindex.
+     *
+     * @param subFiles list of directories to refresh corresponding SearcherManagers
+     * @param host the host address to receive the configuration
+     * @param port the port to use on the host
+     * @throws IOException if an error occurs
+     */
+    public void signalTorefreshSearcherManagers(List<String> subFiles, String host, int port) throws IOException {
+        Message m = Message.createMessage("refresh");
+        m.setText("refresh");
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/configuration/RuntimeEnvironment.java]: Is the text necessary? Add a validation for text field? And possibly create a test for this message testing the validation.

File: src/org/opensolaris/opengrok/configuration/RuntimeEnvironment.java [hunk 3]
```diff
@@ -1172,6 +1172,26 @@ protected void writeConfiguration() throws IOException {
...
+        for (String proj : subFiles) {
+            // subFile entries start with path separator so get basename
+            // to convert them to project names.
+            m.addTag(new File(proj).getName());
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/configuration/RuntimeEnvironment.java]: No need to check if the file actually exists or is a project?

File: src/org/opensolaris/opengrok/configuration/messages/RefreshMessage.java (hunk 1 above)
Comment (vladak) [inline src/org/opensolaris/opengrok/configuration/messages/RefreshMessage.java]: merging now

File: src/org/opensolaris/opengrok/configuration/RuntimeEnvironment.java (hunk 2 above)
Comment (vladak) [inline src/org/opensolaris/opengrok/configuration/RuntimeEnvironment.java]: the text is not needed, will add simple validation+test

File: src/org/opensolaris/opengrok/configuration/RuntimeEnvironment.java (hunk 3 above)
Comment (vladak) [inline src/org/opensolaris/opengrok/configuration/RuntimeEnvironment.java]: For correctness sake, this is checked in `maybeRefreshIndexSearchers()`. Not sure it is worth adding the reply back to the `Messages` tool that no such project exists.

Comment (tulinkry) [inline src/org/opensolaris/opengrok/configuration/RuntimeEnvironment.java]: Ok I see. I don't think there should be a reply

File: src/org/opensolaris/opengrok/configuration/messages/RefreshMessage.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
+/*
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/configuration/messages/RefreshMessage.java]: I don't know why it disappeared anyway write a documentation note about this new type to:
1. Messages.java (usage)
2. README.txt (message types)

I can update the wiki by myself

Comment (vladak) [inline src/org/opensolaris/opengrok/configuration/messages/RefreshMessage.java]: added, although the message is sort of private.

**Pull request merged**